### PR TITLE
Improve bool parsing as XML can support also 0/1 value as bool

### PR DESCRIPTION
**Pull request by punkstarman**
Continuation of #51

### CODE REVIEW

**Comment by punkstarman** [comment]
In #51 a test was requested to complete the feature.

I added a test that deserializes `0` and `1` as boolean. I don't feel that a round-trip test is useful. This is because the `de(ser(x))` round-trip pair is the easiest to test and most useful, but in this case serialization isn't changed.

File: tests/migrated.rs [hunk 1]
```diff
@@ -340,11 +340,14 @@ fn test_parse_u64() {
...
 fn test_parse_bool() {
+    init_logger();
     test_parse_ok(&[
         ("<bla>true</bla>", true),
         ("<bla>false</bla>", false),
         ("<bla> true </bla>", true),
         ("<bla> false </bla>", false),
+        ("<bla>1</bla>", true),
+        ("<bla>0</bla>", false),
     ]);
```
Comment (RReverser) [inline tests/migrated.rs]: Can you also add a test for the error case please?

File: src/de/mod.rs [hunk 2]
```diff
@@ -227,7 +227,28 @@ impl<'de, 'a, R: Read> de::Deserializer<'de> for &'a mut Deserializer<R> {
...
+        }
+        self.read_inner_value::<V, V::Value, _>(|this| {
+            if let XmlEvent::EndElement { .. } = *this.peek()? {
+                return visitor.visit_bool(false);
+            }
+            expect!(this.next()?, XmlEvent::Characters(s) => {
+                match s.as_str() {
+                    "true" | "1" => visitor.visit_bool(true),
+                    "false" | "0" => visitor.visit_bool(false),
+                    _ => Err(ErrorKind::UnexpectedToken(
```
Comment (RReverser) [inline src/de/mod.rs]: Maybe better to use [`Error::unknown_variant`](https://docs.serde.rs/serde/de/trait.Error.html#method.unknown_variant) or [`Error::invalid_value`](https://docs.serde.rs/serde/de/trait.Error.html#method.invalid_value) here.

Comment (punkstarman) [reply src/de/mod.rs]: @RReverser, are you suggesting creating a new `ErrorKind`?

Comment (RReverser) [reply src/de/mod.rs]: Missed this. No, I meant using Serde's own method.

Comment (punkstarman) [reply src/de/mod.rs]: Thanks. I chose `Error::invalid_value`.

File: tests/migrated.rs (hunk 1 above)
Comment (RReverser) [reply tests/migrated.rs]: Ping on this one too.

Comment (RReverser) [reply tests/migrated.rs]: Oh, I see, it was fixed. Nvm.

**Comment by RReverser** [comment]
(approved; as a reminder, please clean up history a bit before merging)

**Pull request merged**